message/messaging_service.cc

```diff
@@ -449,11 +449,9 @@ future<> messaging_service::shutdown() {
 
 future<> messaging_service::stop() {
     if (!_shutting_down) {
-        return shutdown().then([this] {
-            return stop();
```

Comment:
this was lost in the coroutinization

Reply:
Hm... I don't see how :( If `_shutting_down` is not set it co_await-s shutdown, then proceeds with stop. That was the same before the patch

Reply:
Ahh, this is a recursive call. Indeed, we now fall-through to the code previously reached recursively.